File: cometOJ/1473.py
import math
T=int(input())
while T>0:
    n=int(input())
    up=int(math.sqrt(n))
    if up*up==n:
        print("infty")
        T-=1
        continue
    if n%4!=0:
        print("0 0")
    else:
        n/=4
        up=int(math.sqrt(n))
        cnt=0
        ans=0
        for i in range (1,up+1):
            if n%i==0 and n/i+i>up:
                cnt+=1
                ans+=n*(n/i+i)
        print(cnt,int(ans%1000000007))
    T-=1

File: cometOJ/1473.cpp
#include<cstdio>
#include<cmath>
int main()
{
    int T,n;
    scanf("%d",&T);
    while(T--)
    {
        scanf("%d",&n);
        int up=sqrt(n);
        if(up*up==n)
        {
            puts("infty");
            continue;
        }
        if(n%4)
            puts("0 0");
        else
        {
            n/=4;
            up=sqrt(n);
            int cnt=0,ans=0;
            for(int i=1;i<=up;++i)
                if(n%i==0&&n/i+i>up)
                {
                    ++cnt;
                    ans=(ans+(long long)n*(n/i+i)%1000000007)%1000000007;
                }
            printf("%d %d\n",cnt,ans);
        }
    }
    return 0;
}